src/deploy/osp_deployer/deployer.py

```diff
@@ -54,20 +55,42 @@ def get_settings():
                         help='Do not reinstall the Dashboard VM',
                         action='store_true',
                         required=False)
-    parser.add_argument('-validate_settings_only', '--validate_settings_only',
-                        help='Only validate ini and properties files ' +
-                        '(no deployment)',
-                        action='store_true', required=False)
-    args, ignore = parser.parse_known_args()
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('-validate_only', '--validate_only',
+                       help='No deployment - just validate config values',
+                       action='store_true',
+                       required=False)
+    group.add_argument('-tempest_config_only', '--tempest_config_only',
+                       help='Only (re-)generate the tempest.conf file.',
+                       action='store_true', required=False)
+    group.add_argument('-run_tempest_only', '--run_tempest_only',
+                       help='(Re-)generate the tempest.conf file if needed '
+                       'and run tempest.',
+                       action='store_true', required=False)
+
+    args, unknown = parser.parse_known_args()
+    if len(unknown) > 0:
+        parser.print_help()
+        msg = "Invalid argument(s) :"
+        for each in unknown:
+            msg += " " + each + ";"
+        raise AssertionError(msg)
+
+    logger.info("Loading settings file: " + args.settings)
     settings = Settings(args.settings)
-    return settings
+    return settings, args
 
 
 def run_tempest():
-    settings = get_settings()
-    if settings.run_tempest is True:
+    settings, args = get_settings()
+    if settings.run_tempest is True or args.run_tempest_only:
         logger.info("=== Running tempest ==")
         director_vm = Director()
+        is_tempest_conf = director_vm.is_tempest_conf()
```

Comment:
It doesn't look to me like the above line could work because the method in the Director class is _is_tempest_conf().